clippy_lints/src/unitstruct_default_construction.rs

```diff
@@ -0,0 +1,70 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, is_from_proc_macro, match_def_path, paths};
+use hir::{def::Res, ExprKind};
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Check for construction on unit struct using `default`.
+    ///
+    /// ### Why is this bad?
+    /// This adds code complexity and an unnessesary function call.
+    ///
+    /// ### Example
+    /// ```rust
+    /// #[derive(Default)]
+    /// struct S<T> {
+    ///     _marker: PhantomData<T>
+    /// }
+    ///
+    /// let _: S<i32> = S {
+    ///     _marker: PhantomData::default()
+    /// };
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// let _: S<i32> = Something {
+    ///     _marker: PhantomData
+    /// }
+    /// ```
+    #[clippy::version = "1.71.0"]
+    pub UNITSTRUCT_DEFAULT_CONSTRUCTION,
```

Comment:
issue: Not a fan of this name.

@rust-lang/clippy naming thoughts?

`default_constructed_unit` might be fine.

(I don't like `unitstruct`, i think `unit` or `unit_struct` is fine)

Reply:
I went for `default_constructed_unit_struct` for now, to avoid confusion with the unit type `()`. But if a better name comes up, I'm happy to change it again.

Reply:
yeah, that's fine by me. I'll wait for others to chime in